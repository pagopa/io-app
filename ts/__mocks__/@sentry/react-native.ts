export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
