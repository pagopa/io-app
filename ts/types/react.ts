import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
