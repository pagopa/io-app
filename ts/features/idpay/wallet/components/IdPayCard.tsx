import {
  Avatar,
  H3,
  H6,
  LabelSmallAlt,
  VSpacer
} from "@pagopa/io-app-design-system";
import { format } from "date-fns";
import React from "react";
import { ImageURISource, StyleSheet, View } from "react-native";
import WalletCardShape from "../../../../../img/features/idpay/wallet_card.svg";
import I18n from "../../../../i18n";
import { formatNumberAmount } from "../../../../utils/stringBuilder";

export type IdPayCardProps = {
  name: string;
  avatarSource: ImageURISource;
  amount: number;
  expireDate: Date;
};

/**
 * Component that renders the ID PAy card in the wallet
 */
export const IdPayCard = (props: IdPayCardProps) => (
  <View style={styles.container}>
    <View style={styles.card}>
      <WalletCardShape />
    </View>
    <View style={styles.content}>
      <View>
        <View style={styles.header}>
          <H6
            color="blueItalia-850"
            ellipsizeMode="tail"
            numberOfLines={1}
            style={{
              width: "80%"
            }}
          >
            {props.name}
          </H6>
          <Avatar size="small" logoUri={props.avatarSource} />
        </View>
        <VSpacer size={16} />
        <LabelSmallAlt>Disponibile</LabelSmallAlt>
        <H3 color="blueItalia-500">
          {formatNumberAmount(props.amount, true, "right")}
        </H3>
      </View>
      <LabelSmallAlt color="blueItalia-850">
        {I18n.t("bonusCard.validUntil", {
          endDate: format(props.expireDate, "MM/YY")
        })}
      </LabelSmallAlt>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    aspectRatio: 16 / 10
  },
  card: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0
  },
  content: {
    flex: 1,
    paddingTop: 12,
    paddingRight: 12,
    paddingBottom: 16,
    paddingLeft: 16,
    justifyContent: "space-between"
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  }
});
