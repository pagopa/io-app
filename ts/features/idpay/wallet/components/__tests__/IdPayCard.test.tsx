import { render } from "@testing-library/react-native";
import * as React from "react";
import { IdPayCard } from "../IdPayCard";

describe("IdPayCard", () => {
  it("should match the snapshot", () => {
    const component = render(
      <IdPayCard
        name="18 app"
        amount={9999}
        avatarSource={{
          uri: "https://vtlogo.com/wp-content/uploads/2021/08/18app-vector-logo.png"
        }}
        expireDate={new Date(2023, 11, 2)}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
