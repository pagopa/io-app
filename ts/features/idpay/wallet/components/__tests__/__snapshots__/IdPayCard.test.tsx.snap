// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdPayWalletCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "aspectRatio": 1.6,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup
        fill={
          Object {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGDefs>
          <RNSVGClipPath
            fill={
              Object {
                "payload": 4278190080,
                "type": 0,
              }
            }
            name="clip"
          >
            <RNSVGCircle
              cx="75%"
              cy="0"
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              r={16}
            />
            <RNSVGCircle
              cx="75%"
              cy="100%"
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              r={16}
            />
            <RNSVGRect
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              height="100%"
              width="100%"
              x={0}
              y={0}
            />
          </RNSVGClipPath>
        </RNSVGDefs>
        <RNSVGRect
          clipPath="clip"
          fill={
            Object {
              "payload": 4292733946,
              "type": 0,
            }
          }
          height="100%"
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeOpacity",
            ]
          }
          rx={16}
          ry={16}
          stroke={
            Object {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeOpacity={0.12}
          strokeWidth={1}
          width="100%"
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="blueItalia-850"
          defaultColor="black"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
              "lineHeight": 25,
            }
          }
          style={
            Array [
              Object {
                "fontSize": 18,
                "lineHeight": 25,
              },
              Object {
                "color": "#001F3D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          18 app
        </Text>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Array [
              Object {
                "borderColor": "rgba(14,15,19,0.1)",
                "borderCurve": "continuous",
                "borderWidth": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 8,
                "height": 44,
                "padding": 6,
                "width": 44,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderCurve": "continuous",
                  "overflow": "hidden",
                  "resizeMode": "contain",
                },
                Object {
                  "borderRadius": 2,
                },
              ]
            }
          >
            <Image
              accessibilityIgnoresInvertColors={true}
              onError={[Function]}
              source={
                Object {
                  "uri": "https://vtlogo.com/wp-content/uploads/2021/08/18app-vector-logo.png",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 16,
          }
        }
      />
      <Text
        allowFontScaling={false}
        color="black"
        defaultColor="black"
        defaultWeight="SemiBold"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 16,
            "lineHeight": 21,
          }
        }
        style={
          Array [
            Object {
              "fontSize": 16,
              "lineHeight": 21,
            },
            Object {
              "color": "#0E0F13",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "600",
            },
          ]
        }
        weight="SemiBold"
      >
        Disponibile
      </Text>
      <Text
        allowFontScaling={false}
        color="blueItalia-500"
        defaultColor="bluegreyDark"
        defaultWeight="SemiBold"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 24,
            "lineHeight": 34,
          }
        }
        style={
          Array [
            Object {
              "fontSize": 24,
              "lineHeight": 34,
            },
            Object {
              "color": "#0066CC",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "600",
            },
          ]
        }
        weight="SemiBold"
      >
        9,999.00 â‚¬
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      color="blueItalia-850"
      defaultColor="black"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 16,
          "lineHeight": 21,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 16,
            "lineHeight": 21,
          },
          Object {
            "color": "#001F3D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Valida fino al 12/23
    </Text>
  </View>
</View>
`;
